src/segmented_buffer.rs

```diff
@@ -223,15 +224,24 @@ impl Buf for SegmentedBuf<Reusable<Buffer>> {
 
         let mut rem = cnt;
 
-        while rem > 0 {
+        loop {
             let avail = self.bufs[self.read_pos].len() - self.read_offset;
-            if avail >= rem {
+
+            if avail == 0 {
+                break;
```

Comment:
When we are positioned at the end of a chunk, I think we should continue forward to the next chunk, not return.

Reply:
Because of the cnt > 0 check before the loop this case only happens in the rare case that the buffer has been been extended, but nothing written to the final buffer, otherwise avail will always be > 0. It could maybe be changed to if self.buf[self.read_pos].is_empty() ?

Reply:
In fact given the write implementations I don't even think it _can_ be hit